.github/workflows/iris-wallet-desktop.yml

```diff
@@ -166,7 +182,23 @@ jobs:
           python-version: "3.12.3"
 
       - name: Clone rgb-lightning-node repository with submodules
-        run: git clone https://github.com/RGB-Tools/rgb-lightning-node --recurse-submodules --shallow-submodules
+        run: |
+          CURRENT_RLN_NODE_COMMIT=$(grep 'CURRENT_RLN_NODE_COMMIT' ./src/utils/constant.py | awk -F'=' '{print $2}' | tr -d ' "' | xargs)
+
+          if [ -z "$CURRENT_RLN_NODE_COMMIT" ]; then
+            echo "❌ Error: CURRENT_RLN_NODE_COMMIT is empty or not found in constant.py"
+            exit 1
+          fi
+
+          git clone https://github.com/RGB-Tools/rgb-lightning-node --recurse-submodules --shallow-submodules
+          cd rgb-lightning-node
+
+          if ! git checkout ${CURRENT_RLN_NODE_COMMIT}; then
+            echo "❌ Error: Failed to checkout commit ${CURRENT_RLN_NODE_COMMIT}"
+            exit 1
+          fi
+
+          git submodule update --init --recursive
```

Comment:
Could you please avoid duplicating this? Please check https://docs.github.com/en/actions/sharing-automations/avoiding-duplication